eodag/rest/errors.py

```diff
@@ -0,0 +1,172 @@
+# -*- coding: utf-8 -*-
+# Copyright 2024, CS GROUP - France, https://www.csgroup.eu/
+#
+# This file is part of EODAG project
+#     https://www.github.com/CS-SI/EODAG
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+from typing import Dict, List, Tuple, Union
+
+from fastapi import FastAPI, Request
+from fastapi.responses import ORJSONResponse
+from starlette import status
+from starlette.exceptions import HTTPException as StarletteHTTPException
+
+from eodag.rest.types.eodag_search import EODAGSearch
+from eodag.utils.exceptions import (
+    AuthenticationError,
+    DownloadError,
+    EodagError,
+    MisconfiguredError,
+    NoMatchingProductType,
+    NotAvailableError,
+    RequestError,
+    TimeOutError,
+    UnsupportedProductType,
+    UnsupportedProvider,
+    ValidationError,
+)
+
+EODAG_DEFAULT_STATUS_CODES = {
+    AuthenticationError: status.HTTP_500_INTERNAL_SERVER_ERROR,
+    DownloadError: status.HTTP_500_INTERNAL_SERVER_ERROR,
+    MisconfiguredError: status.HTTP_500_INTERNAL_SERVER_ERROR,
+    NotAvailableError: status.HTTP_404_NOT_FOUND,
+    NoMatchingProductType: status.HTTP_404_NOT_FOUND,
+    TimeOutError: status.HTTP_504_GATEWAY_TIMEOUT,
+    UnsupportedProductType: status.HTTP_404_NOT_FOUND,
+    UnsupportedProvider: status.HTTP_404_NOT_FOUND,
+    ValidationError: status.HTTP_400_BAD_REQUEST,
+}
+
+logger = logging.getLogger("eodag.rest.server")
+
+
+class ResponseSearchError(Exception):
+    """Represent a EODAG search error response"""
+
+    def __init__(self, errors: List[Tuple[str, Exception]]) -> None:
+        self._errors = errors
+
+    @property
+    def errors(self) -> List[Dict[str, Union[str, int]]]:
+        """return errors as a list of dict"""
+        error_list: List[Dict[str, Union[str, int]]] = []
+        for name, exception in self._errors:
+
+            error_dict = {"provider": name, "error": exception.__class__.__name__}
+
+            if exception.args:
+                error_dict["message"] = exception.args[0]
+
+            if len(exception.args) > 1:
+                error_dict["detail"] = " ".join(exception.args[1:])
+
+            error_dict["status_code"] = (
+                EODAG_DEFAULT_STATUS_CODES.get(
+                    type(exception), getattr(exception, "status_code", None)
+                )
+                or 500
```

Comment:
Incompatible types in assignment (expression has type "int", target has type "str")mypy[assignment](https://mypy.readthedocs.io/en/stable/_refs.html#code-assignment)